Avoid default values in BButton for size and variant

sfe-efficy opened the pull request:
BEGIN_COMMIT_OVERRIDE
docs: Update BButton to say no variant is apply by default.
fix(BButton): Allow using a BButton with no variant.
END_COMMIT_OVERRIDE

# Describe the PR

Setting as default values a variant as secondary sometimes could be annoying, generating other projects to set an invalid type variant like "empty string" to avoid applying by default the secondary variant. And if they use a checker code like "vue-tsc" that allow to check the templates with strict, it's going to report errors.

The default value for size is redundant and it can avoid.

## PR checklist

<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [X] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [X] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [X] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or have an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

sfe-efficy commented:
About the errors in the "CI", I was running in my local version and I don't see the error.
It's true that all times, I'm getting this one:

docs:build: > vuepress build docs
docs:build: 
docs:build: info Initializing VuePress and preparing data...
docs:build: - Compiling with vite
docs:build: [vite:resolve] Missing "./preload-helper" export in "vite" package
docs:build: × Compiling with vite - failed
docs:build: Error: Missing "./preload-helper" export in "vite" package

sfe-efficy commented:
> About the errors in the "CI", I was running in my local version and I don't see the error. It's true that all times, I'm getting this one:
> 
> docs:build: > vuepress build docs docs:build: docs:build: info Initializing VuePress and preparing data... docs:build: - Compiling with vite docs:build: [vite:resolve] Missing "./preload-helper" export in "vite" package docs:build: × Compiling with vite - failed docs:build: Error: Missing "./preload-helper" export in "vite" package

Fixed.
I was using yarn v1, instead of yarn v3, and I installed the plugin for workspaces.

https://backstage.io/docs/tutorials/yarn-migration

VividLemon commented:
The library doesn't use yarn lmao

sfe-efficy commented:
> The library doesn't use yarn lmao

Thanks for your helpful reply xD ....

I notice after seeing a lot of new files to delete.
Ok, I see that I should use pnpm tool, but I return to have the same error with vite during the build.

VividLemon commented:
You should be opening the package at the _root_ level, and running _pnpm install_ and _pnpm dev_ from the root.

VividLemon commented on packages/bootstrap-vue-3/src/components/BInputGroup/BInputGroup.vue:
```diff
@@ -28,7 +28,7 @@ interface BInputGroupProps {
   prepend?: string
   prependHtml?: string
-  size?: 'sm' | 'lg' // Exclude<InputSize, 'md'>
+  size?: Exclude<InputSize, 'md'>
```
https://github.com/cdmoro/bootstrap-vue-3/pull/801#issuecomment-1311077812

> Originally, the thought was that many components could keep their defaults, where "undefined" would give them their defaults. However, the idea was to introduce "null" as a possibility, to create the ability to remove the item outright. But again, BV was written a bit weirdly, along with Vue 2, in the regards of the two being pretty much identical, rather then having separate uses.

This is another case of what I meant by "complex types". Doing this is not allowed in script setup syntax. The comment marker is there explaining what the complex type is, for the future, but it's there for a reason, because it's **impossible** syntax at the moment. 

This is why you're getting an error.

sfe-efficy replied:
Hmm, I was testing with other code and it looks no more error.
I've just push a commit with a code that it's working.

Basically, I tried to define the type in the InputSize.d.ts file instead of doing it in the vue component file.

VividLemon replied:
I know, and it's neat that it works, but it's kind of a lot of code and boilerplate to get things working. It's cleaner to have it just contain a note for the future once it does become fully supported in Vue

sfe-efficy commented:
> You should be opening the package at the _root_ level, and running _pnpm install_ and _pnpm dev_ from the root of the workspace.

Yes, I'm executing the pnpm install from root package :/

│  ERROR  run failed: command  exited (1)
└─ Failed in 58.1s
 ELIFECYCLE  Command failed with exit code 1.
PS C:\Some\Path\To\My\RootFolder\bootstrap-vue-3> pnpm install

Other commands like pnpm test, dev and so on are ok

VividLemon commented on packages/bootstrap-vue-3/src/components/BInputGroup/BInputGroup.vue:
```diff
@@ -28,7 +28,7 @@ interface BInputGroupProps {
   prepend?: string
   prependHtml?: string
-  size?: Exclude<InputSize, 'md'>
+  size?: InputSizeOnlySmallOrLarge
```
Please just revert this back to the original.

sfe-efficy replied:
Ok, I have the feeling that InputSize is not the best name for this type, because we are defining the basic size types, but for group input is only available two options only

VividLemon replied:
The extended code that you wrote will work, and it's neat that it works. It's just a lot of boilerplate for something that will eventually be replaced when it's a fully supported feature in Vue.

VividLemon commented:
> > You should be opening the package at the _root_ level, and running _pnpm install_ and _pnpm dev_ from the root of the workspace.
> 
> Yes, I'm executing the pnpm install from root package :/
> 
> │ ERROR run failed: command exited (1) └─ Failed in 58.1s  ELIFECYCLE  Command failed with exit code 1. PS C:\Some\Path\To\My\RootFolder\bootstrap-vue-3> pnpm install
> 
> Other commands like pnpm test, dev and so on are ok

You could make a video screen share recording and show me, I've never seen that error before.

sfe-efficy commented:
> > > You should be opening the package at the _root_ level, and running _pnpm install_ and _pnpm dev_ from the root of the workspace.
> > 
> > 
> > Yes, I'm executing the pnpm install from root package :/
> > │ ERROR run failed: command exited (1) └─ Failed in 58.1s  ELIFECYCLE  Command failed with exit code 1. PS C:\Some\Path\To\My\RootFolder\bootstrap-vue-3> pnpm install
> > Other commands like pnpm test, dev and so on are ok
> 
> You could make a video screen share recording and show me, I've never seen that error before.

Here is the video
https://www.loom.com/share/29de00edfd454f849010e0539ed9a2b7

VividLemon commented:
> > > > You should be opening the package at the _root_ level, and running _pnpm install_ and _pnpm dev_ from the root of the workspace.
> > > 
> > > 
> > > Yes, I'm executing the pnpm install from root package :/
> > > │ ERROR run failed: command exited (1) └─ Failed in 58.1s  ELIFECYCLE  Command failed with exit code 1. PS C:\Some\Path\To\My\RootFolder\bootstrap-vue-3> pnpm install
> > > Other commands like pnpm test, dev and so on are ok
> > 
> > 
> > You could make a video screen share recording and show me, I've never seen that error before.
> 
> Here is the video https://www.loom.com/share/29de00edfd454f849010e0539ed9a2b7

Try syncing with the main branch upstream. (fetch on github in your private repo, then git pull on dev). Then delete all node_modules & pnpm-lock , then retry

sfe-efficy commented:
> > > > > You should be opening the package at the _root_ level, and running _pnpm install_ and _pnpm dev_ from the root of the workspace.
> > > > 
> > > > 
> > > > Yes, I'm executing the pnpm install from root package :/
> > > > │ ERROR run failed: command exited (1) └─ Failed in 58.1s  ELIFECYCLE  Command failed with exit code 1. PS C:\Some\Path\To\My\RootFolder\bootstrap-vue-3> pnpm install
> > > > Other commands like pnpm test, dev and so on are ok
> > > 
> > > 
> > > You could make a video screen share recording and show me, I've never seen that error before.
> > 
> > 
> > Here is the video https://www.loom.com/share/29de00edfd454f849010e0539ed9a2b7
> 
> Try syncing with the main branch upstream. (fetch on github in your private repo, then git pull on dev). Then delete all node_modules & pnpm-lock , then retry

Fixed :) I removed node_modules and retry and it had work.
Thanks!

VividLemon commented:
Is there anything I'm waiting on before I merge this?

sfe-efficy commented:
> Is there anything I'm waiting on before I merge this?

No really, I finished last Saturday. The override commits are not displaying the reality, I guess is the only pending part.

VividLemon commented on packages/bootstrap-vue-3/src/components/BPopover.vue:
```diff
@@ -61,7 +53,6 @@ export default defineComponent({
     delay: {type: [Number, Object] as PropType<number | BPopoverDelayObject>, default: 0},
     triggers: {type: String as PropType<Popover.Options['trigger']>, default: 'click'},
     show: {type: [Boolean, String] as PropType<Booleanish>, default: false},
-    variant: {type: String as PropType<ColorVariant>, default: undefined},
```
Why was variant here removed

VividLemon commented on packages/bootstrap-vue-3/src/types/RoundedTypes.d.ts:
```diff
@@ -0,0 +1,5 @@
+type RoundedNumberStringTypes = "0" | "1" | "2" | "3" | "4" | "5";
+// empty string is considered as true value
+type RoundedStringTypes = "bottom" | "circle" | "end" | "pill" | "start" | "top" | "";
+export type RoundedTypesAll = boolean | RoundedNumberTypes | RoundedStringTypes | RoundedNumberStringTypes
```
```suggestion
export type RoundedTypesAll = boolean | RoundedNumberTypes | RoundedStringTypes | RoundedNumberStringTypes

```

VividLemon commented on packages/bootstrap-vue-3/src/utils/colors.ts:
```diff
@@ -0,0 +1,3 @@
+import {BgColorVariant} from '../types'
+
+export const textVariantDarkTypes: BgColorVariant[] = ['warning', 'info', 'light']
```
```suggestion
export const textVariantDarkTypes: BgColorVariant[] = ['warning', 'info', 'light']

```

VividLemon commented:
Please resolve the merge conflict and fetch the upstream branch, as well as answer the few questions I have posted in the review.

VividLemon commented:
Going to close this and revisit implementing this at a later time. For now there's more important things to do. Will be done prior to 1.0
